redis/src/bloom.rs

```diff
@@ -0,0 +1,310 @@
+    Size,
+    /// Return the number of sub-filters
+    Filters,
+    /// Return he number of items inserted in the filter
+    Items,
+    /// Return the expansion rate of the filter
+    Expansion,
+}
+
+impl ToRedisArgs for InfoType {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        match *self {
+            InfoType::Capacity => {
+                out.write_arg(b"CAPACITY");
+            }
+            InfoType::Size => {
+                out.write_arg(b"SIZE");
+            }
+            InfoType::Filters => {
+                out.write_arg(b"FILTERS");
+            }
+            InfoType::Expansion => {
+                out.write_arg(b"EXPANSION");
+            }
+            InfoType::Items => {
+                out.write_arg(b"ITEMS");
+            }
+        }
+    }
+}
+
+/// Response returned by the `bf_info_all` command.
+/// See the [Redis documentation](https://redis.io/docs/latest/commands/bf.info/) for more information.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct AllInfoResponse {
+    /// The number of unique items that can be stored in this Bloom filter before scaling would be required
+    pub capacity: i64,
+    /// The memory size: number of bytes allocated for this Bloom filter.
+    pub memory_size: i64,
+    /// Return the number of sub-filters
+    pub number_of_filters: i64,
+    /// The number of items inserted in the filter
+    pub number_of_items_inserted: i64,
+    /// The expansion rate of the filter (if filter is scaling)
+    pub expansion_rate: Option<i64>,
+}
+
+impl FromRedisValue for AllInfoResponse {
+    fn from_redis_value(v: &Value) -> RedisResult<Self> {
+        let hm: RedisResult<HashMap<String, Option<_>>> = from_redis_value(v);
+        hm.map(|hm| {
+            let resp = &mut AllInfoResponse {
+                capacity: 0,
+                memory_size: 0,
+                number_of_filters: 0,
+                number_of_items_inserted: 0,
+                expansion_rate: None,
+            };
+
+            hm.iter().for_each(|(k, v)| match k.as_str() {
+                "Capacity" => resp.capacity = v.map_or(0, |x| x),
+                "Size" => resp.memory_size = v.map_or(0, |x| x),
+                "Number of filters" => resp.number_of_filters = v.map_or(0, |x| x),
+                "Number of items inserted" => resp.number_of_items_inserted = v.map_or(0, |x| x),
+                "Expansion rate" => resp.expansion_rate = *v,
+                _ => { /*quietly ignore unknown keys*/ }
+            });
+            *resp
+        })
+    }
+}
+/// Response returned by the bf_info command (used to return a specific field).
+/// See the [Redis documentation](https://redis.io/docs/latest/commands/bf.info/) for more information.
+///
+#[derive(Debug)]
+pub struct SingleInfoResponse {
+    /// The value of the requested field if exists
+    pub value: Option<i64>,
+}
+
+macro_rules! not_convertible_error {
```

Comment:
instead of copying this from acl.rs, can you move it to a shared scope?

Reply:
not so sure of my fix here. I moved it to redis/macros. but then the test compilation started complaing about unused macro. so I added a #[allow(unused_macros)].